verified-broadcast/assets/setup.private.js

```diff
@@ -0,0 +1,164 @@
+async function updateEnvironmentVariable(twilioClient, context, key, value) {
+  if (!context.SERVICE_SID && context.ENVIRONMENT_SID) {
+    context[key] = value;
+    process.env[key] = value;
+    return;
+  }
+  if (context[key] === value) {
+    // variable already exists
+  } else if (context[key]) {
+    const existingVariables = await twilioClient.serverless
+      .services(context.SERVICE_SID)
+      .environments(context.ENVIRONMENT_SID)
+      .variables.list();
+    const [variableToUpdate] = existingVariables.filter(
+      (variable) => variable.key === key
+    );
+    variableToUpdate.update({ value });
+    context[key] = value;
+    process.env[key] = value;
+  } else {
+    await twilioClient.serverless
+      .services(context.SERVICE_SID)
+      .environments(context.ENVIRONMENT_SID)
+      .variables.create({
+        key,
+        value,
+      });
+    context[key] = value;
+    process.env[key] = value;
+  }
+}
+
+async function getMessagingServiceFromPhoneNumber(
+  twilioClient,
+  targetPhoneNumber
+) {
+  const messagingServices = await twilioClient.messaging.services.list({
+    limit: 2000,
+  });
+
+  const phoneNumberMap = new Map();
+  const requests = messagingServices.map(async (service) => {
+    const numbers = await service.phoneNumbers.list({ limit: 2000 });
+    numbers.forEach(({ phoneNumber }) => {
+      phoneNumberMap.set(phoneNumber, service.sid);
+    });
+    return true;
+  });
+  await Promise.all(requests);
+
+  if (phoneNumberMap.has(targetPhoneNumber)) {
+    return phoneNumberMap.get(targetPhoneNumber);
+  }
+  return null;
+}
+
+async function setupMessagingService(twilioClient, context) {
+  if (!context.TWILIO_PHONE_NUMBER) {
+    throw new Error(
+      'Cannot create a Messaging Service without a TWILIO_PHONE_NUMBER'
+    );
+  }
+
+  const existingServiceSid = await getMessagingServiceFromPhoneNumber(
+    twilioClient,
+    context.TWILIO_PHONE_NUMBER
+  );
+  if (existingServiceSid !== null) {
+    await updateEnvironmentVariable(
+      twilioClient,
+      context,
+      'MESSAGING_SERVICE_SID',
+      existingServiceSid
+    );
+    return context.MESSAGING_SERVICE_SID;
+  }
+
+  const { sid } = await twilioClient.messaging.services.create({
+    friendlyName: 'Verified Broadcast Messaging Service',
+    useInboundWebhookOnNumber: true,
```

Comment:
What happens if the existing messaging service doesn't have this setting?

Reply:
It doesn't matter because they already configured a messaging service to their liking. This only makes sure I don't break phone numbers that might already have a webhook configured.